components/servo/lib.rs

```diff
@@ -372,6 +372,14 @@ impl<Window> Servo<Window> where Window: WindowMethods + 'static {
 
                 (_, ShutdownState::ShuttingDown) => {},
 
+                (EmbedderMsg::KeyEvent(top_level_browsing_context, ch, key, state, modified),
+                 ShutdownState::NotShuttingDown) => {
+                    if state == KeyState::Pressed {
+                        let msg = EmbedderMsg::KeyEvent(top_level_browsing_context, ch, key, state, modified);
```

Comment:
Is there a way to not re-construct the event here?

Reply:
If we did not have to check the state, we could match `(msg @ EmbedderMsg::KeyEvent(..), ShutdownState::NotShuttingDown)`; otherwise we would need to match a clone of the message and push the original.